source/mir/las/sum.d

```diff
@@ -1782,18 +1782,56 @@ nothrow unittest
         tuple([2.0^^53+10.0, 1.0, 2.0^^-100], 2.0^^53+12.0),
         tuple([2.0^^53-4.0, 0.5, 2.0^^-54], 2.0^^53-3.0),
         tuple([M-2.0^^970, -1, M], 1.7976931348623157e+308),
-        tuple([double.max, double.max*2.^^-54], double.max),
-        tuple([double.max, double.max*2.^^-53], double.infinity),
         tuple(iota(1, 1001).map!(a => 1.0/a).array , 7.4854708605503451),
         tuple(iota(1, 1001).map!(a => (-1.0)^^a/a).array, -0.69264743055982025), //0.693147180559945309417232121458176568075500134360255254120680...
         tuple(iota(1, 1001).map!(a => 1.0/a).retro.array , 7.4854708605503451),
         tuple(iota(1, 1001).map!(a => (-1.0)^^a/a).retro.array, -0.69264743055982025),
+        tuple([-double.infinity, 1e308, 1e308, -double.infinity], -double.infinity),
+        tuple([-2.^^970, M, M, -2.^^-1074], double.max),
+        tuple([-M, -M, 2.^^970, 2.^^-1074], -double.max),
+        tuple([2.^^930, -2.^^980, M, M, M, -M], 1.7976931348622137e+308),
+        tuple([M, M, -1e307], 1.6976931348623159e+308),
+        tuple([1e16, 1., 1e-16], 10000000000000002.0),
+    ];
+    foreach (i, test; tests)
+    {
+        foreach (t; test[0]) summator.put(t);
+        auto r = test[1];
+        auto s = summator.sum;
+        version(X86)
+        {
+            assert(summator.isNaN() == r.isNaN());
+            assert(summator.isFinite() == r.isFinite() || r == -double.max && s == -double.infinity || r == double.max && s == double.infinity);
+            assert(summator.isInfinity() == r.isInfinity() || r == -double.max && s == -double.infinity || r == double.max && s == double.infinity);
+            assert(nextDown(s) <= r && r <= nextUp(s) || s.isNaN && r.isNaN);
+        }
+        else
+        {
+            assert(summator.isNaN() == r.isNaN());
+            assert(summator.isFinite() == r.isFinite());
+            assert(summator.isInfinity() == r.isInfinity());
+            assert(s == r || s.isNaN && r.isNaN);
+        }
+        summator = 0.0;
+    }
+}
+
+nothrow unittest
```

Comment:
I splitted the tests infinity tests out and run them for the pairwise summator too. As we can't just create an array of them, `runTests` was created.